docs/tracing.md

```diff
@@ -0,0 +1,306 @@
+* [Overview](#overview)
+* [Introduction](#introduction)
+* [OpenTelemetry summary](#opentelemetry-summary)
+* [Architecture](#architecture)
+    * [Runtime tracing architecture](#runtime-tracing-architecture)
+    * [Agent tracing architecture](#agent-tracing-architecture)
+* [Agent tracing terminology](#agent-tracing-terminology)
+    * [Trace modes](#trace-modes)
+    * [Trace types](#trace-types)
+* [Prerequisites](#prerequisites)
+* [Enable tracing](#enable-tracing)
+    * [Enable runtime tracing](#enable-runtime-tracing)
+    * [Enable agent tracing](#enable-agent-tracing)
+        * [Enable static agent tracing](#enable-static-agent-tracing)
+        * [Enable dynamic agent tracing](#enable-dynamic-agent-tracing)
+* [Appendices](#appendices)
+    * [Agent tracing requirements](#agent-tracing-requirements)
+        * [Host environment](#host-environment)
+        * [Guest environment](#guest-environment)
+    * [Agent tracing limitations](#agent-tracing-limitations)
+    * [Performance impact](#performance-impact)
+    * [Agent shutdown behaviour](#agent-shutdown-behaviour)
+    * [Set up a tracing development environment](#set-up-a-tracing-development-environment)
+
+# Overview
+
+This document explains how to trace Kata Containers components.
+
+> **Warning:**
+>
+> This document explains the planned set of tracing features. However, tracing
+> support in the agent is still under development. Initially,
+> [_static isolated_ tracing is being added](https://github.com/kata-containers/kata-containers/pull/1929).
+> The remaining trace work items are tracked in the
+> [tracing GitHub project](https://github.com/orgs/kata-containers/projects/28).
+
+# Introduction
+
+The Kata Containers runtime and agent are able to generate
+[OpenTelemetry][opentelemetry] trace spans, which allow the administrator to
+observe what they are doing and how much time they are spending on each
+activity.
+
+# OpenTelemetry summary
+
+An OpenTelemetry-enabled application creates a number of trace "spans". A span
+contains the following attributes:
+
+- A name
+- A pair of timestamps (recording the start time and end time of some operation)
+- A reference to the span's parent span
+
+All spans need to be *finished*, or *completed*, to allow the OpenTelemetry
+framework to generate the final trace information (by effectively closing the
+transaction encompassing the initial (root) span and all its children).
+
+# Architecture
+
+## Runtime tracing architecture
+
+The runtime, which runs in the host environment, has been modified to generate
+trace spans which are sent to a trace collector on the host.
+
+## Agent tracing architecture
+
+The Kata Containers agent runs inside a virtual machine (VM). An OpenTelemetry
+system (such as [Jaeger][jaeger-tracing]) uses a collector to gather up trace
+spans for viewing and processing.
```

Comment:
I don't understand if the last sentence is trying to describe something specific to Kata, or if it's just a general observation about OpenTelemetry. If the latter, maybe "OpenTelemetry systems", and then a quick note on how the collector is installed / started on the host.